env/work/srw_python/uti_plot_matplotlib.py

```diff
@@ -104,10 +104,44 @@ def uti_plot2d1d(self, data, x_range, y_range, xc, yc, labels, _graphs_joined=Tr
         return self._maybe_savefig(fig)
 
     #def srw_ascii_plot(fname):
-    def uti_data_file_plot(self, _fname, _read_labels=1, _e=0, _x=0, _y=0, _graphs_joined=1, _traj_report=False, _traj_axis='x'):
+    def uti_data_file_plot(self, _fname, _read_labels=1, _e=0, _x=0, _y=0, _graphs_joined=1,
+                           _multicolumn_data=False, _column_x=None, _column_y=None, #MR31102017
+                           _scale='linear', _width_pixels=None):
         #data, mode, allrange = srw_ascii_load(fname)
         #data, mode, allrange, arLabels, arUnits = _file_load(_fname, _read_labels)
-        data, mode, allrange, arLabels, arUnits = uti_plot_com.file_load(_fname, _read_labels, _traj_report, _traj_axis)
+        data, mode, allrange, arLabels, arUnits = uti_plot_com.file_load(_fname, _read_labels, _multicolumn_data) #MR31102017
+        if not _multicolumn_data:
+            data = np.array(data)
+        if mode == 3:
+            try:
+                fwhm_dict = uti_math.fwhm(np.linspace(allrange[0], allrange[1], allrange[2]), data, return_as_dict=True)
+            except:
+                fwhm_dict = {'fwhm': -1}
+            print('FWHM: {:.5f} [{}]'.format(fwhm_dict['fwhm'], arUnits[0]))
+        if _scale != 'linear':
+            available_scales = ['log', 'log2', 'log10']
+            if _scale not in available_scales:
+                raise ValueError('Scale "{}" is not supported. Available scales: {}.'.format(_scale, ', '.join(available_scales)))
+            data[np.where(data <= 0.)] = 1.e-23
+            data = getattr(np, _scale)(data)
+        if _width_pixels:
+            try:
+                from scipy.ndimage import zoom
+                data = np.reshape(data, (allrange[5], allrange[8]), order='C')
+                resize_factor = float(_width_pixels) / float(allrange[5])
+                print('Size before: {}  Dimensions: {}'.format(data.size, data.shape))
+                data = zoom(data, resize_factor)
+                if _scale == 'linear':
+                    data[np.where(data < 0.)] = 0.0
+                print('Size after : {}  Dimensions: {}'.format(data.size, data.shape))
+                allrange = list(allrange)
+                allrange[5] = data.shape[0]
+                allrange[8] = data.shape[1]
+                allrange = tuple(allrange)
+                data = np.reshape(data, (data.shape[0] * data.shape[1]), order='C')
+            except:
+                print('Cannot resize the image - scipy.ndimage.zoom() cannot be imported.')
+                pass
```

Comment:
OK, updated this; though it is not clear which exactly lines had to be deleted (I made a guess).

Reply:
I think I only added the code, didn't remove anything in this place since there are only green lines with `+` symbols, no red color with `-`.